[mdoc] Tool breaks when reflecting .NET assemblies

The [current tool release](https://github.com/mono/api-doc-tools/releases/tag/mdoc-5.3.5.2) breaks when trying to reflect .NET assemblies, with:

```
2017-12-15T22:51:09.8019075Z Updating: System.Windows.DataFormats
2017-12-15T22:51:09.8203926Z Updating: System.Windows.DataFormat
2017-12-15T22:51:09.8314512Z Updating: System.Windows.DataObject
2017-12-15T22:51:10.1778991Z ##[error]mdoc: System.NotSupportedException: Unsupported member type: Mono.Documentation.Util.AttachedEventReference
   at Mono.Documentation.Updater.VBFullMemberFormatter.IsSupported(MemberReference mref)
   at Mono.Documentation.Updater.MemberFormatter.GetDeclaration(MemberReference mreference)
   at Mono.Documentation.MDocUpdater.UpdateMember(DocsNodeInfo info, FrameworkTypeEntry typeEntry, Dictionary`2 implementedMembers)
   at Mono.Documentation.MDocUpdater.MakeMember(XmlDocument doc, DocsNodeInfo info, XmlNode members, FrameworkTypeEntry typeEntry, Dictionary`2 iplementedMembers)
   at Mono.Documentation.MDocUpdater.DoUpdateType2(String message, XmlDocument basefile, TypeDefinition type, FrameworkTypeEntry typeEntry, String output, Boolean insertSince)
   at Mono.Documentation.MDocUpdater.DoUpdateType(TypeDefinition type, FrameworkTypeEntry typeEntry, String basepath, String dest)
   at Mono.Documentation.MDocUpdater.DoUpdateAssembly(AssemblySet assemblySet, AssemblyDefinition assembly, XmlElement index_types, String source, String dest, HashSet`1 goodfiles)
   at Mono.Documentation.MDocUpdater.DoUpdateAssemblies(String source, String dest)
   at Mono.Documentation.MDocUpdater.Run(IEnumerable`1 args)
   at Mono.Documentation.MDoc.Run(String[] args)
   at Mono.Documentation.MDoc.Main(String[] args)
See `mdoc help' for more information or use --debug to diagnose.
```

--- a/mdoc/Consts.cs
+++ b/mdoc/Consts.cs
@@ -3,7 +3,7 @@ namespace Mono.Documentation
 {
 	public static class Consts
 	{
-		public static string MonoVersion = "5.3.5.2";
+		public static string MonoVersion = "5.3.5.3";
 		public const string DocId = "DocId";
 		public const string VbNet = "VB.NET";
 		public const string DocIdLowCase = "docid";

--- a/mdoc/Makefile
+++ b/mdoc/Makefile
@@ -450,7 +450,7 @@ check-monodocer-import-fx-update: check-monodocer-import-fx-work
 check-monodocer-attached-entities:
 	-rm -Rf Test/en.actual
 	msbuild ../mdoc/Test/AttachedEventsAndProperties/AttachedEventsAndProperties.csproj -property:Configuration=Release
-	$(MONO) $(PROGRAM) update -o Test/en.actual Test/AttachedEventsAndProperties/bin/Release/AttachedEventsAndProperties.dll -lang docid
+	$(MONO) $(PROGRAM) update -o Test/en.actual Test/AttachedEventsAndProperties/bin/Release/AttachedEventsAndProperties.dll -lang docid -lang vb.net -lang fsharp
 	$(DIFF) Test/en.expected-attached-entities Test/en.actual
 	
 Test/TestClass.dll:

--- a/mdoc/Mono.Documentation/Updater/Formatters/VBFullMemberFormatter.cs
+++ b/mdoc/Mono.Documentation/Updater/Formatters/VBFullMemberFormatter.cs
@@ -775,6 +775,14 @@ namespace Mono.Documentation.Updater
                     && IsSupportedNaming(@event);
             }
 
+            var @event2 = mref as AttachedEventReference;
+            if (@event2 != null)
+                return true;
+
+            var prop = mref as AttachedPropertyReference;
+            if (prop != null)
+                return true;
+
             throw new NotSupportedException("Unsupported member type: " + mref.GetType().FullName);
         }
 

--- a/mdoc/Test/en.expected-attached-entities/AttachedEventsAndProperties/AquariumFilter.xml
+++ b/mdoc/Test/en.expected-attached-entities/AttachedEventsAndProperties/AquariumFilter.xml
@@ -2,6 +2,8 @@
   <TypeSignature Language="C#" Value="public class AquariumFilter" />
   <TypeSignature Language="ILAsm" Value=".class public auto ansi beforefieldinit AquariumFilter extends System.Object" />
   <TypeSignature Language="DocId" Value="T:AttachedEventsAndProperties.AquariumFilter" />
+  <TypeSignature Language="VB.NET" Value="Public Class AquariumFilter" />
+  <TypeSignature Language="F#" Value="type AquariumFilter = class" />
   <AssemblyInfo>
     <AssemblyName>AttachedEventsAndProperties</AssemblyName>
     <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -19,6 +21,7 @@
       <MemberSignature Language="C#" Value="public AquariumFilter ();" />
       <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
       <MemberSignature Language="DocId" Value="M:AttachedEventsAndProperties.AquariumFilter.#ctor" />
+      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
       <MemberType>Constructor</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -33,6 +36,8 @@
       <MemberSignature Language="C#" Value="public static System.Windows.RoutedEvent NeedsCleaningEvent { get; set; }" />
       <MemberSignature Language="ILAsm" Value=".property class System.Windows.RoutedEvent NeedsCleaningEvent" />
       <MemberSignature Language="DocId" Value="P:AttachedEventsAndProperties.AquariumFilter.NeedsCleaningEvent" />
+      <MemberSignature Language="VB.NET" Value="Public Shared Property NeedsCleaningEvent As RoutedEvent" />
+      <MemberSignature Language="F#" Value="member this.NeedsCleaningEvent : System.Windows.RoutedEvent with get, set" Usage="AttachedEventsAndProperties.AquariumFilter.NeedsCleaningEvent" />
       <MemberType>Property</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>

--- a/mdoc/Test/en.expected-attached-entities/AttachedEventsAndProperties/AquariumObject.xml
+++ b/mdoc/Test/en.expected-attached-entities/AttachedEventsAndProperties/AquariumObject.xml
@@ -2,6 +2,8 @@
   <TypeSignature Language="C#" Value="public class AquariumObject" />
   <TypeSignature Language="ILAsm" Value=".class public auto ansi beforefieldinit AquariumObject extends System.Object" />
   <TypeSignature Language="DocId" Value="T:AttachedEventsAndProperties.AquariumObject" />
+  <TypeSignature Language="VB.NET" Value="Public Class AquariumObject" />
+  <TypeSignature Language="F#" Value="type AquariumObject = class" />
   <AssemblyInfo>
     <AssemblyName>AttachedEventsAndProperties</AssemblyName>
     <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -19,6 +21,7 @@
       <MemberSignature Language="C#" Value="public AquariumObject ();" />
       <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
       <MemberSignature Language="DocId" Value="M:AttachedEventsAndProperties.AquariumObject.#ctor" />
+      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
       <MemberType>Constructor</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>

--- a/mdoc/Test/en.expected-attached-entities/AttachedEventsAndProperties/AttachedEventExample.xml
+++ b/mdoc/Test/en.expected-attached-entities/AttachedEventsAndProperties/AttachedEventExample.xml
@@ -2,6 +2,8 @@
   <TypeSignature Language="C#" Value="public class AttachedEventExample" />
   <TypeSignature Language="ILAsm" Value=".class public auto ansi beforefieldinit AttachedEventExample extends System.Object" />
   <TypeSignature Language="DocId" Value="T:AttachedEventsAndProperties.AttachedEventExample" />
+  <TypeSignature Language="VB.NET" Value="Public Class AttachedEventExample" />
+  <TypeSignature Language="F#" Value="type AttachedEventExample = class" />
   <AssemblyInfo>
     <AssemblyName>AttachedEventsAndProperties</AssemblyName>
     <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -19,6 +21,7 @@
       <MemberSignature Language="C#" Value="public AttachedEventExample ();" />
       <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
       <MemberSignature Language="DocId" Value="M:AttachedEventsAndProperties.AttachedEventExample.#ctor" />
+      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
       <MemberType>Constructor</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -33,6 +36,8 @@
       <MemberSignature Language="C#" Value="public static void AddDragOverHandler (System.Windows.DependencyObject element, System.Windows.DragEventHandler handler);" />
       <MemberSignature Language="ILAsm" Value=".method public static hidebysig void AddDragOverHandler(class System.Windows.DependencyObject element, class System.Windows.DragEventHandler handler) cil managed" />
       <MemberSignature Language="DocId" Value="M:AttachedEventsAndProperties.AttachedEventExample.AddDragOverHandler(System.Windows.DependencyObject,System.Windows.DragEventHandler)" />
+      <MemberSignature Language="VB.NET" Value="Public Shared Sub AddDragOverHandler (element As DependencyObject, handler As DragEventHandler)" />
+      <MemberSignature Language="F#" Value="static member AddDragOverHandler : System.Windows.DependencyObject * System.Windows.DragEventHandler -&gt; unit" Usage="AttachedEventsAndProperties.AttachedEventExample.AddDragOverHandler (element, handler)" />
       <MemberType>Method</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -55,6 +60,8 @@
       <MemberSignature Language="C#" Value="public static void AddNeedsCleaning2Handler (System.Windows.DependencyObject d, System.Windows.RoutedEventHandler handler);" />
       <MemberSignature Language="ILAsm" Value=".method public static hidebysig void AddNeedsCleaning2Handler(class System.Windows.DependencyObject d, class System.Windows.RoutedEventHandler handler) cil managed" />
       <MemberSignature Language="DocId" Value="M:AttachedEventsAndProperties.AttachedEventExample.AddNeedsCleaning2Handler(System.Windows.DependencyObject,System.Windows.RoutedEventHandler)" />
+      <MemberSignature Language="VB.NET" Value="Public Shared Sub AddNeedsCleaning2Handler (d As DependencyObject, handler As RoutedEventHandler)" />
+      <MemberSignature Language="F#" Value="static member AddNeedsCleaning2Handler : System.Windows.DependencyObject * System.Windows.RoutedEventHandler -&gt; unit" Usage="AttachedEventsAndProperties.AttachedEventExample.AddNeedsCleaning2Handler (d, handler)" />
       <MemberType>Method</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -77,6 +84,8 @@
       <MemberSignature Language="C#" Value="protected static void AddNeedsCleaning4Handler (System.Windows.DependencyObject d, System.Windows.RoutedEventHandler handler);" />
       <MemberSignature Language="ILAsm" Value=".method familystatic hidebysig void AddNeedsCleaning4Handler(class System.Windows.DependencyObject d, class System.Windows.RoutedEventHandler handler) cil managed" />
       <MemberSignature Language="DocId" Value="M:AttachedEventsAndProperties.AttachedEventExample.AddNeedsCleaning4Handler(System.Windows.DependencyObject,System.Windows.RoutedEventHandler)" />
+      <MemberSignature Language="VB.NET" Value="Protected Shared Sub AddNeedsCleaning4Handler (d As DependencyObject, handler As RoutedEventHandler)" />
+      <MemberSignature Language="F#" Value="static member AddNeedsCleaning4Handler : System.Windows.DependencyObject * System.Windows.RoutedEventHandler -&gt; unit" Usage="AttachedEventsAndProperties.AttachedEventExample.AddNeedsCleaning4Handler (d, handler)" />
       <MemberType>Method</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -99,6 +108,8 @@
       <MemberSignature Language="C#" Value="public void AddNeedsCleaning5Handler (System.Windows.DependencyObject d, System.Windows.RoutedEventHandler handler);" />
       <MemberSignature Language="ILAsm" Value=".method public hidebysig instance void AddNeedsCleaning5Handler(class System.Windows.DependencyObject d, class System.Windows.RoutedEventHandler handler) cil managed" />
       <MemberSignature Language="DocId" Value="M:AttachedEventsAndProperties.AttachedEventExample.AddNeedsCleaning5Handler(System.Windows.DependencyObject,System.Windows.RoutedEventHandler)" />
+      <MemberSignature Language="VB.NET" Value="Public Sub AddNeedsCleaning5Handler (d As DependencyObject, handler As RoutedEventHandler)" />
+      <MemberSignature Language="F#" Value="member this.AddNeedsCleaning5Handler : System.Windows.DependencyObject * System.Windows.RoutedEventHandler -&gt; unit" Usage="attachedEventExample.AddNeedsCleaning5Handler (d, handler)" />
       <MemberType>Method</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -121,6 +132,8 @@
       <MemberSignature Language="C#" Value="public static void AddNeedsCleaning6Handler (System.Windows.DependencyObject d, System.Windows.RoutedEventHandler handler);" />
       <MemberSignature Language="ILAsm" Value=".method public static hidebysig void AddNeedsCleaning6Handler(class System.Windows.DependencyObject d, class System.Windows.RoutedEventHandler handler) cil managed" />
       <MemberSignature Language="DocId" Value="M:AttachedEventsAndProperties.AttachedEventExample.AddNeedsCleaning6Handler(System.Windows.DependencyObject,System.Windows.RoutedEventHandler)" />
+      <MemberSignature Language="VB.NET" Value="Public Shared Sub AddNeedsCleaning6Handler (d As DependencyObject, handler As RoutedEventHandler)" />
+      <MemberSignature Language="F#" Value="static member AddNeedsCleaning6Handler : System.Windows.DependencyObject * System.Windows.RoutedEventHandler -&gt; unit" Usage="AttachedEventsAndProperties.AttachedEventExample.AddNeedsCleaning6Handler (d, handler)" />
       <MemberType>Method</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -143,6 +156,8 @@
       <MemberSignature Language="C#" Value="public static void AddNeedsCleaning7Handler (System.Windows.DependencyObject d, System.Windows.RoutedEventHandler handler);" />
       <MemberSignature Language="ILAsm" Value=".method public static hidebysig void AddNeedsCleaning7Handler(class System.Windows.DependencyObject d, class System.Windows.RoutedEventHandler handler) cil managed" />
       <MemberSignature Language="DocId" Value="M:AttachedEventsAndProperties.AttachedEventExample.AddNeedsCleaning7Handler(System.Windows.DependencyObject,System.Windows.RoutedEventHandler)" />
+      <MemberSignature Language="VB.NET" Value="Public Shared Sub AddNeedsCleaning7Handler (d As DependencyObject, handler As RoutedEventHandler)" />
+      <MemberSignature Language="F#" Value="static member AddNeedsCleaning7Handler : System.Windows.DependencyObject * System.Windows.RoutedEventHandler -&gt; unit" Usage="AttachedEventsAndProperties.AttachedEventExample.AddNeedsCleaning7Handler (d, handler)" />
       <MemberType>Method</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -165,6 +180,8 @@
       <MemberSignature Language="C#" Value="public static void AddNeedsCleaningHandler (System.Windows.DependencyObject d, System.Windows.RoutedEventHandler handler);" />
       <MemberSignature Language="ILAsm" Value=".method public static hidebysig void AddNeedsCleaningHandler(class System.Windows.DependencyObject d, class System.Windows.RoutedEventHandler handler) cil managed" />
       <MemberSignature Language="DocId" Value="M:AttachedEventsAndProperties.AttachedEventExample.AddNeedsCleaningHandler(System.Windows.DependencyObject,System.Windows.RoutedEventHandler)" />
+      <MemberSignature Language="VB.NET" Value="Public Shared Sub AddNeedsCleaningHandler (d As DependencyObject, handler As RoutedEventHandler)" />
+      <MemberSignature Language="F#" Value="static member AddNeedsCleaningHandler : System.Windows.DependencyObject * System.Windows.RoutedEventHandler -&gt; unit" Usage="AttachedEventsAndProperties.AttachedEventExample.AddNeedsCleaningHandler (d, handler)" />
       <MemberType>Method</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -187,6 +204,8 @@
       <MemberSignature Language="C#" Value="see AddDragOverHandler, and RemoveDragOverHandler" />
       <MemberSignature Language="ILAsm" Value="see AddDragOverHandler, and RemoveDragOverHandler" />
       <MemberSignature Language="DocId" Value="E:AttachedEventsAndProperties.AttachedEventExample.DragOver" />
+      <MemberSignature Language="VB.NET" Value="see AddDragOverHandler, and RemoveDragOverHandler" />
+      <MemberSignature Language="F#" Value="see AddDragOverHandler, and RemoveDragOverHandler" Usage="see AddDragOverHandler, and RemoveDragOverHandler" />
       <MemberType>AttachedEvent</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -200,6 +219,8 @@
       <MemberSignature Language="C#" Value="public static readonly System.Windows.RoutedEvent DragOverEvent;" />
       <MemberSignature Language="ILAsm" Value=".field public static initonly class System.Windows.RoutedEvent DragOverEvent" />
       <MemberSignature Language="DocId" Value="F:AttachedEventsAndProperties.AttachedEventExample.DragOverEvent" />
+      <MemberSignature Language="VB.NET" Value="Public Shared ReadOnly DragOverEvent As RoutedEvent " />
+      <MemberSignature Language="F#" Value=" staticval mutable DragOverEvent : System.Windows.RoutedEvent" Usage="AttachedEventsAndProperties.AttachedEventExample.DragOverEvent" />
       <MemberType>Field</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -216,6 +237,8 @@
       <MemberSignature Language="C#" Value="public static readonly System.Windows.RoutedEvent E;" />
       <MemberSignature Language="ILAsm" Value=".field public static initonly class System.Windows.RoutedEvent E" />
       <MemberSignature Language="DocId" Value="F:AttachedEventsAndProperties.AttachedEventExample.E" />
+      <MemberSignature Language="VB.NET" Value="Public Shared ReadOnly E As RoutedEvent " />
+      <MemberSignature Language="F#" Value=" staticval mutable E : System.Windows.RoutedEvent" Usage="AttachedEventsAndProperties.AttachedEventExample.E" />
       <MemberType>Field</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -232,6 +255,8 @@
       <MemberSignature Language="C#" Value="see AddNeedsCleaningHandler, and RemoveNeedsCleaningHandler" />
       <MemberSignature Language="ILAsm" Value="see AddNeedsCleaningHandler, and RemoveNeedsCleaningHandler" />
       <MemberSignature Language="DocId" Value="E:AttachedEventsAndProperties.AttachedEventExample.NeedsCleaning" />
+      <MemberSignature Language="VB.NET" Value="see AddNeedsCleaningHandler, and RemoveNeedsCleaningHandler" />
+      <MemberSignature Language="F#" Value="see AddNeedsCleaningHandler, and RemoveNeedsCleaningHandler" Usage="see AddNeedsCleaningHandler, and RemoveNeedsCleaningHandler" />
       <MemberType>AttachedEvent</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -245,6 +270,8 @@
       <MemberSignature Language="C#" Value="public static readonly System.Windows.RoutedEvent NeedsCleaning2Event;" />
       <MemberSignature Language="ILAsm" Value=".field public static initonly class System.Windows.RoutedEvent NeedsCleaning2Event" />
       <MemberSignature Language="DocId" Value="F:AttachedEventsAndProperties.AttachedEventExample.NeedsCleaning2Event" />
+      <MemberSignature Language="VB.NET" Value="Public Shared ReadOnly NeedsCleaning2Event As RoutedEvent " />
+      <MemberSignature Language="F#" Value=" staticval mutable NeedsCleaning2Event : System.Windows.RoutedEvent" Usage="AttachedEventsAndProperties.AttachedEventExample.NeedsCleaning2Event" />
       <MemberType>Field</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -261,6 +288,8 @@
       <MemberSignature Language="C#" Value="public static readonly System.Windows.RoutedEvent NeedsCleaning3Event;" />
       <MemberSignature Language="ILAsm" Value=".field public static initonly class System.Windows.RoutedEvent NeedsCleaning3Event" />
       <MemberSignature Language="DocId" Value="F:AttachedEventsAndProperties.AttachedEventExample.NeedsCleaning3Event" />
+      <MemberSignature Language="VB.NET" Value="Public Shared ReadOnly NeedsCleaning3Event As RoutedEvent " />
+      <MemberSignature Language="F#" Value=" staticval mutable NeedsCleaning3Event : System.Windows.RoutedEvent" Usage="AttachedEventsAndProperties.AttachedEventExample.NeedsCleaning3Event" />
       <MemberType>Field</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -277,6 +306,8 @@
       <MemberSignature Language="C#" Value="public static readonly System.Windows.RoutedEvent NeedsCleaning4Event;" />
       <MemberSignature Language="ILAsm" Value=".field public static initonly class System.Windows.RoutedEvent NeedsCleaning4Event" />
       <MemberSignature Language="DocId" Value="F:AttachedEventsAndProperties.AttachedEventExample.NeedsCleaning4Event" />
+      <MemberSignature Language="VB.NET" Value="Public Shared ReadOnly NeedsCleaning4Event As RoutedEvent " />
+      <MemberSignature Language="F#" Value=" staticval mutable NeedsCleaning4Event : System.Windows.RoutedEvent" Usage="AttachedEventsAndProperties.AttachedEventExample.NeedsCleaning4Event" />
       <MemberType>Field</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -293,6 +324,8 @@
       <MemberSignature Language="C#" Value="public readonly System.Windows.RoutedEvent NeedsCleaning5Event;" />
       <MemberSignature Language="ILAsm" Value=".field public initonly class System.Windows.RoutedEvent NeedsCleaning5Event" />
       <MemberSignature Language="DocId" Value="F:AttachedEventsAndProperties.AttachedEventExample.NeedsCleaning5Event" />
+      <MemberSignature Language="VB.NET" Value="Public ReadOnly NeedsCleaning5Event As RoutedEvent " />
+      <MemberSignature Language="F#" Value="val mutable NeedsCleaning5Event : System.Windows.RoutedEvent" Usage="AttachedEventsAndProperties.AttachedEventExample.NeedsCleaning5Event" />
       <MemberType>Field</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -309,6 +342,8 @@
       <MemberSignature Language="C#" Value="public static readonly System.Windows.RoutedEvent NeedsCleaning6Event6;" />
       <MemberSignature Language="ILAsm" Value=".field public static initonly class System.Windows.RoutedEvent NeedsCleaning6Event6" />
       <MemberSignature Language="DocId" Value="F:AttachedEventsAndProperties.AttachedEventExample.NeedsCleaning6Event6" />
+      <MemberSignature Language="VB.NET" Value="Public Shared ReadOnly NeedsCleaning6Event6 As RoutedEvent " />
+      <MemberSignature Language="F#" Value=" staticval mutable NeedsCleaning6Event6 : System.Windows.RoutedEvent" Usage="AttachedEventsAndProperties.AttachedEventExample.NeedsCleaning6Event6" />
       <MemberType>Field</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -325,6 +360,8 @@
       <MemberSignature Language="C#" Value="public static readonly System.Windows.RoutedEvent NeedsCleaningEvent;" />
       <MemberSignature Language="ILAsm" Value=".field public static initonly class System.Windows.RoutedEvent NeedsCleaningEvent" />
       <MemberSignature Language="DocId" Value="F:AttachedEventsAndProperties.AttachedEventExample.NeedsCleaningEvent" />
+      <MemberSignature Language="VB.NET" Value="Public Shared ReadOnly NeedsCleaningEvent As RoutedEvent " />
+      <MemberSignature Language="F#" Value=" staticval mutable NeedsCleaningEvent : System.Windows.RoutedEvent" Usage="AttachedEventsAndProperties.AttachedEventExample.NeedsCleaningEvent" />
       <MemberType>Field</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -341,6 +378,8 @@
       <MemberSignature Language="C#" Value="public static void RemoveDragOverHandler (System.Windows.DependencyObject element, System.Windows.DragEventHandler handler);" />
       <MemberSignature Language="ILAsm" Value=".method public static hidebysig void RemoveDragOverHandler(class System.Windows.DependencyObject element, class System.Windows.DragEventHandler handler) cil managed" />
       <MemberSignature Language="DocId" Value="M:AttachedEventsAndProperties.AttachedEventExample.RemoveDragOverHandler(System.Windows.DependencyObject,System.Windows.DragEventHandler)" />
+      <MemberSignature Language="VB.NET" Value="Public Shared Sub RemoveDragOverHandler (element As DependencyObject, handler As DragEventHandler)" />
+      <MemberSignature Language="F#" Value="static member RemoveDragOverHandler : System.Windows.DependencyObject * System.Windows.DragEventHandler -&gt; unit" Usage="AttachedEventsAndProperties.AttachedEventExample.RemoveDragOverHandler (element, handler)" />
       <MemberType>Method</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -363,6 +402,8 @@
       <MemberSignature Language="C#" Value="public static void RemoveNeedsCleaning3Handler (System.Windows.DependencyObject d, System.Windows.RoutedEventHandler handler);" />
       <MemberSignature Language="ILAsm" Value=".method public static hidebysig void RemoveNeedsCleaning3Handler(class System.Windows.DependencyObject d, class System.Windows.RoutedEventHandler handler) cil managed" />
       <MemberSignature Language="DocId" Value="M:AttachedEventsAndProperties.AttachedEventExample.RemoveNeedsCleaning3Handler(System.Windows.DependencyObject,System.Windows.RoutedEventHandler)" />
+      <MemberSignature Language="VB.NET" Value="Public Shared Sub RemoveNeedsCleaning3Handler (d As DependencyObject, handler As RoutedEventHandler)" />
+      <MemberSignature Language="F#" Value="static member RemoveNeedsCleaning3Handler : System.Windows.DependencyObject * System.Windows.RoutedEventHandler -&gt; unit" Usage="AttachedEventsAndProperties.AttachedEventExample.RemoveNeedsCleaning3Handler (d, handler)" />
       <MemberType>Method</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -385,6 +426,8 @@
       <MemberSignature Language="C#" Value="protected static void RemoveNeedsCleaning4Handler (System.Windows.DependencyObject d, System.Windows.RoutedEventHandler handler);" />
       <MemberSignature Language="ILAsm" Value=".method familystatic hidebysig void RemoveNeedsCleaning4Handler(class System.Windows.DependencyObject d, class System.Windows.RoutedEventHandler handler) cil managed" />
       <MemberSignature Language="DocId" Value="M:AttachedEventsAndProperties.AttachedEventExample.RemoveNeedsCleaning4Handler(System.Windows.DependencyObject,System.Windows.RoutedEventHandler)" />
+      <MemberSignature Language="VB.NET" Value="Protected Shared Sub RemoveNeedsCleaning4Handler (d As DependencyObject, handler As RoutedEventHandler)" />
+      <MemberSignature Language="F#" Value="static member RemoveNeedsCleaning4Handler : System.Windows.DependencyObject * System.Windows.RoutedEventHandler -&gt; unit" Usage="AttachedEventsAndProperties.AttachedEventExample.RemoveNeedsCleaning4Handler (d, handler)" />
       <MemberType>Method</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -407,6 +450,8 @@
       <MemberSignature Language="C#" Value="public void RemoveNeedsCleaning5Handler (System.Windows.DependencyObject d, System.Windows.RoutedEventHandler handler);" />
       <MemberSignature Language="ILAsm" Value=".method public hidebysig instance void RemoveNeedsCleaning5Handler(class System.Windows.DependencyObject d, class System.Windows.RoutedEventHandler handler) cil managed" />
       <MemberSignature Language="DocId" Value="M:AttachedEventsAndProperties.AttachedEventExample.RemoveNeedsCleaning5Handler(System.Windows.DependencyObject,System.Windows.RoutedEventHandler)" />
+      <MemberSignature Language="VB.NET" Value="Public Sub RemoveNeedsCleaning5Handler (d As DependencyObject, handler As RoutedEventHandler)" />
+      <MemberSignature Language="F#" Value="member this.RemoveNeedsCleaning5Handler : System.Windows.DependencyObject * System.Windows.RoutedEventHandler -&gt; unit" Usage="attachedEventExample.RemoveNeedsCleaning5Handler (d, handler)" />
       <MemberType>Method</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -429,6 +474,8 @@
       <MemberSignature Language="C#" Value="public static void RemoveNeedsCleaning6Handler (System.Windows.DependencyObject d, System.Windows.RoutedEventHandler handler);" />
       <MemberSignature Language="ILAsm" Value=".method public static hidebysig void RemoveNeedsCleaning6Handler(class System.Windows.DependencyObject d, class System.Windows.RoutedEventHandler handler) cil managed" />
       <MemberSignature Language="DocId" Value="M:AttachedEventsAndProperties.AttachedEventExample.RemoveNeedsCleaning6Handler(System.Windows.DependencyObject,System.Windows.RoutedEventHandler)" />
+      <MemberSignature Language="VB.NET" Value="Public Shared Sub RemoveNeedsCleaning6Handler (d As DependencyObject, handler As RoutedEventHandler)" />
+      <MemberSignature Language="F#" Value="static member RemoveNeedsCleaning6Handler : System.Windows.DependencyObject * System.Windows.RoutedEventHandler -&gt; unit" Usage="AttachedEventsAndProperties.AttachedEventExample.RemoveNeedsCleaning6Handler (d, handler)" />
       <MemberType>Method</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -451,6 +498,8 @@
       <MemberSignature Language="C#" Value="public static void RemoveNeedsCleaning7Handler (System.Windows.DependencyObject d, System.Windows.RoutedEventHandler handler);" />
       <MemberSignature Language="ILAsm" Value=".method public static hidebysig void RemoveNeedsCleaning7Handler(class System.Windows.DependencyObject d, class System.Windows.RoutedEventHandler handler) cil managed" />
       <MemberSignature Language="DocId" Value="M:AttachedEventsAndProperties.AttachedEventExample.RemoveNeedsCleaning7Handler(System.Windows.DependencyObject,System.Windows.RoutedEventHandler)" />
+      <MemberSignature Language="VB.NET" Value="Public Shared Sub RemoveNeedsCleaning7Handler (d As DependencyObject, handler As RoutedEventHandler)" />
+      <MemberSignature Language="F#" Value="static member RemoveNeedsCleaning7Handler : System.Windows.DependencyObject * System.Windows.RoutedEventHandler -&gt; unit" Usage="AttachedEventsAndProperties.AttachedEventExample.RemoveNeedsCleaning7Handler (d, handler)" />
       <MemberType>Method</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -473,6 +522,8 @@
       <MemberSignature Language="C#" Value="public static void RemoveNeedsCleaningHandler (System.Windows.DependencyObject d, System.Windows.RoutedEventHandler handler);" />
       <MemberSignature Language="ILAsm" Value=".method public static hidebysig void RemoveNeedsCleaningHandler(class System.Windows.DependencyObject d, class System.Windows.RoutedEventHandler handler) cil managed" />
       <MemberSignature Language="DocId" Value="M:AttachedEventsAndProperties.AttachedEventExample.RemoveNeedsCleaningHandler(System.Windows.DependencyObject,System.Windows.RoutedEventHandler)" />
+      <MemberSignature Language="VB.NET" Value="Public Shared Sub RemoveNeedsCleaningHandler (d As DependencyObject, handler As RoutedEventHandler)" />
+      <MemberSignature Language="F#" Value="static member RemoveNeedsCleaningHandler : System.Windows.DependencyObject * System.Windows.RoutedEventHandler -&gt; unit" Usage="AttachedEventsAndProperties.AttachedEventExample.RemoveNeedsCleaningHandler (d, handler)" />
       <MemberType>Method</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>

--- a/mdoc/Test/en.expected-attached-entities/AttachedEventsAndProperties/AttachedPropertyExample.xml
+++ b/mdoc/Test/en.expected-attached-entities/AttachedEventsAndProperties/AttachedPropertyExample.xml
@@ -2,6 +2,8 @@
   <TypeSignature Language="C#" Value="public static class AttachedPropertyExample" />
   <TypeSignature Language="ILAsm" Value=".class public auto ansi abstract sealed beforefieldinit AttachedPropertyExample extends System.Object" />
   <TypeSignature Language="DocId" Value="T:AttachedEventsAndProperties.AttachedPropertyExample" />
+  <TypeSignature Language="VB.NET" Value="Public Class AttachedPropertyExample" />
+  <TypeSignature Language="F#" Value="type AttachedPropertyExample = class" />
   <AssemblyInfo>
     <AssemblyName>AttachedEventsAndProperties</AssemblyName>
     <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -19,6 +21,8 @@
       <MemberSignature Language="C#" Value="public static bool GetIsBubbleSource (System.Windows.UIElement element);" />
       <MemberSignature Language="ILAsm" Value=".method public static hidebysig bool GetIsBubbleSource(class System.Windows.UIElement element) cil managed" />
       <MemberSignature Language="DocId" Value="M:AttachedEventsAndProperties.AttachedPropertyExample.GetIsBubbleSource(System.Windows.UIElement)" />
+      <MemberSignature Language="VB.NET" Value="Public Shared Function GetIsBubbleSource (element As UIElement) As Boolean" />
+      <MemberSignature Language="F#" Value="static member GetIsBubbleSource : System.Windows.UIElement -&gt; bool" Usage="AttachedEventsAndProperties.AttachedPropertyExample.GetIsBubbleSource element" />
       <MemberType>Method</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -40,6 +44,8 @@
       <MemberSignature Language="C#" Value="public static bool GetIsBubbleSource3 (System.Windows.UIElement element);" />
       <MemberSignature Language="ILAsm" Value=".method public static hidebysig bool GetIsBubbleSource3(class System.Windows.UIElement element) cil managed" />
       <MemberSignature Language="DocId" Value="M:AttachedEventsAndProperties.AttachedPropertyExample.GetIsBubbleSource3(System.Windows.UIElement)" />
+      <MemberSignature Language="VB.NET" Value="Public Shared Function GetIsBubbleSource3 (element As UIElement) As Boolean" />
+      <MemberSignature Language="F#" Value="static member GetIsBubbleSource3 : System.Windows.UIElement -&gt; bool" Usage="AttachedEventsAndProperties.AttachedPropertyExample.GetIsBubbleSource3 element" />
       <MemberType>Method</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -61,6 +67,8 @@
       <MemberSignature Language="C#" Value="see GetIsBubbleSource, and SetIsBubbleSource" />
       <MemberSignature Language="ILAsm" Value="see GetIsBubbleSource, and SetIsBubbleSource" />
       <MemberSignature Language="DocId" Value="P:AttachedEventsAndProperties.AttachedPropertyExample.IsBubbleSource" />
+      <MemberSignature Language="VB.NET" Value="see GetIsBubbleSource, and SetIsBubbleSource" />
+      <MemberSignature Language="F#" Value="see GetIsBubbleSource, and SetIsBubbleSource" Usage="see GetIsBubbleSource, and SetIsBubbleSource" />
       <MemberType>AttachedProperty</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -74,6 +82,8 @@
       <MemberSignature Language="C#" Value="public static readonly System.Windows.DependencyProperty IsBubbleSource2Property;" />
       <MemberSignature Language="ILAsm" Value=".field public static initonly class System.Windows.DependencyProperty IsBubbleSource2Property" />
       <MemberSignature Language="DocId" Value="F:AttachedEventsAndProperties.AttachedPropertyExample.IsBubbleSource2Property" />
+      <MemberSignature Language="VB.NET" Value="Public Shared ReadOnly IsBubbleSource2Property As DependencyProperty " />
+      <MemberSignature Language="F#" Value=" staticval mutable IsBubbleSource2Property : System.Windows.DependencyProperty" Usage="AttachedEventsAndProperties.AttachedPropertyExample.IsBubbleSource2Property" />
       <MemberType>Field</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -90,6 +100,8 @@
       <MemberSignature Language="C#" Value="public static readonly System.Windows.DependencyProperty IsBubbleSourceProperty;" />
       <MemberSignature Language="ILAsm" Value=".field public static initonly class System.Windows.DependencyProperty IsBubbleSourceProperty" />
       <MemberSignature Language="DocId" Value="F:AttachedEventsAndProperties.AttachedPropertyExample.IsBubbleSourceProperty" />
+      <MemberSignature Language="VB.NET" Value="Public Shared ReadOnly IsBubbleSourceProperty As DependencyProperty " />
+      <MemberSignature Language="F#" Value=" staticval mutable IsBubbleSourceProperty : System.Windows.DependencyProperty" Usage="AttachedEventsAndProperties.AttachedPropertyExample.IsBubbleSourceProperty" />
       <MemberType>Field</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -106,6 +118,8 @@
       <MemberSignature Language="C#" Value="public static readonly System.Windows.DependencyProperty P;" />
       <MemberSignature Language="ILAsm" Value=".field public static initonly class System.Windows.DependencyProperty P" />
       <MemberSignature Language="DocId" Value="F:AttachedEventsAndProperties.AttachedPropertyExample.P" />
+      <MemberSignature Language="VB.NET" Value="Public Shared ReadOnly P As DependencyProperty " />
+      <MemberSignature Language="F#" Value=" staticval mutable P : System.Windows.DependencyProperty" Usage="AttachedEventsAndProperties.AttachedPropertyExample.P" />
       <MemberType>Field</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -122,6 +136,8 @@
       <MemberSignature Language="C#" Value="public static void SetIsBubbleSource (System.Windows.UIElement element, bool value);" />
       <MemberSignature Language="ILAsm" Value=".method public static hidebysig void SetIsBubbleSource(class System.Windows.UIElement element, bool value) cil managed" />
       <MemberSignature Language="DocId" Value="M:AttachedEventsAndProperties.AttachedPropertyExample.SetIsBubbleSource(System.Windows.UIElement,System.Boolean)" />
+      <MemberSignature Language="VB.NET" Value="Public Shared Sub SetIsBubbleSource (element As UIElement, value As Boolean)" />
+      <MemberSignature Language="F#" Value="static member SetIsBubbleSource : System.Windows.UIElement * bool -&gt; unit" Usage="AttachedEventsAndProperties.AttachedPropertyExample.SetIsBubbleSource (element, value)" />
       <MemberType>Method</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -144,6 +160,8 @@
       <MemberSignature Language="C#" Value="public static void SetIsBubbleSource2 (System.Windows.UIElement element, bool value);" />
       <MemberSignature Language="ILAsm" Value=".method public static hidebysig void SetIsBubbleSource2(class System.Windows.UIElement element, bool value) cil managed" />
       <MemberSignature Language="DocId" Value="M:AttachedEventsAndProperties.AttachedPropertyExample.SetIsBubbleSource2(System.Windows.UIElement,System.Boolean)" />
+      <MemberSignature Language="VB.NET" Value="Public Shared Sub SetIsBubbleSource2 (element As UIElement, value As Boolean)" />
+      <MemberSignature Language="F#" Value="static member SetIsBubbleSource2 : System.Windows.UIElement * bool -&gt; unit" Usage="AttachedEventsAndProperties.AttachedPropertyExample.SetIsBubbleSource2 (element, value)" />
       <MemberType>Method</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -166,6 +184,8 @@
       <MemberSignature Language="C#" Value="public static void SetIsBubbleSource3 (System.Windows.UIElement element, bool value);" />
       <MemberSignature Language="ILAsm" Value=".method public static hidebysig void SetIsBubbleSource3(class System.Windows.UIElement element, bool value) cil managed" />
       <MemberSignature Language="DocId" Value="M:AttachedEventsAndProperties.AttachedPropertyExample.SetIsBubbleSource3(System.Windows.UIElement,System.Boolean)" />
+      <MemberSignature Language="VB.NET" Value="Public Shared Sub SetIsBubbleSource3 (element As UIElement, value As Boolean)" />
+      <MemberSignature Language="F#" Value="static member SetIsBubbleSource3 : System.Windows.UIElement * bool -&gt; unit" Usage="AttachedEventsAndProperties.AttachedPropertyExample.SetIsBubbleSource3 (element, value)" />
       <MemberType>Method</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>

--- a/mdoc/Test/en.expected-attached-entities/System.Windows/DragEventArgs.xml
+++ b/mdoc/Test/en.expected-attached-entities/System.Windows/DragEventArgs.xml
@@ -2,6 +2,8 @@
   <TypeSignature Language="C#" Value="public sealed class DragEventArgs : System.Windows.RoutedEventArgs" />
   <TypeSignature Language="ILAsm" Value=".class public auto ansi sealed beforefieldinit DragEventArgs extends System.Windows.RoutedEventArgs" />
   <TypeSignature Language="DocId" Value="T:System.Windows.DragEventArgs" />
+  <TypeSignature Language="VB.NET" Value="Public NotInheritable Class DragEventArgs&#xA;Inherits RoutedEventArgs" />
+  <TypeSignature Language="F#" Value="type DragEventArgs = class&#xA;    inherit RoutedEventArgs" />
   <AssemblyInfo>
     <AssemblyName>AttachedEventsAndProperties</AssemblyName>
     <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -19,6 +21,7 @@
       <MemberSignature Language="C#" Value="public DragEventArgs ();" />
       <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
       <MemberSignature Language="DocId" Value="M:System.Windows.DragEventArgs.#ctor" />
+      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
       <MemberType>Constructor</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>

--- a/mdoc/Test/en.expected-attached-entities/System.Windows/DragEventHandler.xml
+++ b/mdoc/Test/en.expected-attached-entities/System.Windows/DragEventHandler.xml
@@ -2,6 +2,8 @@
   <TypeSignature Language="C#" Value="public delegate void DragEventHandler(object sender, DragEventArgs e);" />
   <TypeSignature Language="ILAsm" Value=".class public auto ansi sealed DragEventHandler extends System.MulticastDelegate" />
   <TypeSignature Language="DocId" Value="T:System.Windows.DragEventHandler" />
+  <TypeSignature Language="VB.NET" Value="Public Delegate Sub DragEventHandler(sender As Object, e As DragEventArgs)" />
+  <TypeSignature Language="F#" Value="type DragEventHandler = delegate of obj * DragEventArgs -&gt; unit" />
   <AssemblyInfo>
     <AssemblyName>AttachedEventsAndProperties</AssemblyName>
     <AssemblyVersion>1.0.0.0</AssemblyVersion>

--- a/mdoc/Test/en.expected-attached-entities/System.Windows/RoutedEvent.xml
+++ b/mdoc/Test/en.expected-attached-entities/System.Windows/RoutedEvent.xml
@@ -2,6 +2,8 @@
   <TypeSignature Language="C#" Value="public class RoutedEvent" />
   <TypeSignature Language="ILAsm" Value=".class public auto ansi beforefieldinit RoutedEvent extends System.Object" />
   <TypeSignature Language="DocId" Value="T:System.Windows.RoutedEvent" />
+  <TypeSignature Language="VB.NET" Value="Public Class RoutedEvent" />
+  <TypeSignature Language="F#" Value="type RoutedEvent = class" />
   <AssemblyInfo>
     <AssemblyName>AttachedEventsAndProperties</AssemblyName>
     <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -19,6 +21,7 @@
       <MemberSignature Language="C#" Value="public RoutedEvent ();" />
       <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
       <MemberSignature Language="DocId" Value="M:System.Windows.RoutedEvent.#ctor" />
+      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
       <MemberType>Constructor</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>

--- a/mdoc/Test/en.expected-attached-entities/System.Windows/RoutedEventArgs.xml
+++ b/mdoc/Test/en.expected-attached-entities/System.Windows/RoutedEventArgs.xml
@@ -2,6 +2,8 @@
   <TypeSignature Language="C#" Value="public class RoutedEventArgs : EventArgs" />
   <TypeSignature Language="ILAsm" Value=".class public auto ansi beforefieldinit RoutedEventArgs extends System.EventArgs" />
   <TypeSignature Language="DocId" Value="T:System.Windows.RoutedEventArgs" />
+  <TypeSignature Language="VB.NET" Value="Public Class RoutedEventArgs&#xA;Inherits EventArgs" />
+  <TypeSignature Language="F#" Value="type RoutedEventArgs = class&#xA;    inherit EventArgs" />
   <AssemblyInfo>
     <AssemblyName>AttachedEventsAndProperties</AssemblyName>
     <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -19,6 +21,7 @@
       <MemberSignature Language="C#" Value="public RoutedEventArgs ();" />
       <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
       <MemberSignature Language="DocId" Value="M:System.Windows.RoutedEventArgs.#ctor" />
+      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
       <MemberType>Constructor</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>

--- a/mdoc/Test/en.expected-attached-entities/System.Windows/RoutedEventHandler.xml
+++ b/mdoc/Test/en.expected-attached-entities/System.Windows/RoutedEventHandler.xml
@@ -2,6 +2,8 @@
   <TypeSignature Language="C#" Value="public delegate void RoutedEventHandler(object sender, RoutedEventArgs e);" />
   <TypeSignature Language="ILAsm" Value=".class public auto ansi sealed RoutedEventHandler extends System.MulticastDelegate" />
   <TypeSignature Language="DocId" Value="T:System.Windows.RoutedEventHandler" />
+  <TypeSignature Language="VB.NET" Value="Public Delegate Sub RoutedEventHandler(sender As Object, e As RoutedEventArgs)" />
+  <TypeSignature Language="F#" Value="type RoutedEventHandler = delegate of obj * RoutedEventArgs -&gt; unit" />
   <AssemblyInfo>
     <AssemblyName>AttachedEventsAndProperties</AssemblyName>
     <AssemblyVersion>1.0.0.0</AssemblyVersion>

--- a/mdoc/Test/en.expected-attached-entities/System.Windows/UIElement.xml
+++ b/mdoc/Test/en.expected-attached-entities/System.Windows/UIElement.xml
@@ -2,6 +2,8 @@
   <TypeSignature Language="C#" Value="public class UIElement : System.Windows.DependencyObject" />
   <TypeSignature Language="ILAsm" Value=".class public auto ansi beforefieldinit UIElement extends System.Windows.DependencyObject" />
   <TypeSignature Language="DocId" Value="T:System.Windows.UIElement" />
+  <TypeSignature Language="VB.NET" Value="Public Class UIElement&#xA;Inherits DependencyObject" />
+  <TypeSignature Language="F#" Value="type UIElement = class&#xA;    inherit DependencyObject" />
   <AssemblyInfo>
     <AssemblyName>AttachedEventsAndProperties</AssemblyName>
     <AssemblyVersion>1.0.0.0</AssemblyVersion>
@@ -19,6 +21,7 @@
       <MemberSignature Language="C#" Value="public UIElement ();" />
       <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
       <MemberSignature Language="DocId" Value="M:System.Windows.UIElement.#ctor" />
+      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
       <MemberType>Constructor</MemberType>
       <AssemblyInfo>
         <AssemblyVersion>1.0.0.0</AssemblyVersion>

--- a/mdoc/mdoc.nuspec
+++ b/mdoc/mdoc.nuspec
@@ -2,7 +2,7 @@
 <package >
   <metadata>
     <id>mdoc</id>
-    <version>5.3.5.2</version>
+    <version>5.3.5.3</version>
     <title>mdoc</title>
     <authors>Microsoft</authors>
     <owners>Microsoft</owners>

